operantanalysis/operantanalysis.py

```diff
@@ -42,11 +42,10 @@ def rewardretrieval(timecode, eventcode):
     :param eventcode: list of event codes from operant conditioning file
     :return: number of reinforcers (dippers) presented, number retrieved, and latency to retrieve as floats
     """
-    dipperon = [i for i in range(len(eventcode)) if eventcode[i] == 25]
-    dipperoff = [i for i in range(len(eventcode)) if eventcode[i] == 26]
-    pokeon = [i for i in range(len(eventcode)) if eventcode[i] == 1011]
-    pokeoff = [i for i in range(len(eventcode)) if eventcode[i] == 1001]
-
+    dipperon = [i for i, event in enumerate(eventcode) if event == 25]
+    dipperoff = [i for i, event in enumerate(eventcode) if event == 26]
+    pokeon = [i for i, event in enumerate(eventcode) if event == 1011]
+    pokeoff = [i for i, event in enumerate(eventcode) if event == 1001]
```

Comment:
such long variable names make it a bit hard to read the code.  Also not yet sure if you wouldn't be off separating codes into some dictionary ,e.g. events = {1011: 'pokeon', 1001: 'pokeoff', ...} and then just mapping your evencodes accordingly and manipulating that list directly instead of having two nested loops below. may be will get a better clue later
but in the loop below I also see no reason to e.g. use range(len -- since you use those [i] [x] only to get that specific value, so why not `for diponidx  in dipperon` etc?

Reply:
I added a dictionary of eventcodes so the code is more readable for people who are not me now. I originally used range(len because I needed to find the matching dipper on and dipper off based on their position in the two lists, I'm keeping that for now, but hopefully I can learn about pandas and that may become a better way to deal with this data.